src/app/compare/compare.component.ts

```diff
@@ -83,13 +87,66 @@ export class CompareComponent implements AfterViewInit, OnInit {
 
   ngAfterViewInit(): void {
     if (this.compareData) {
-      this.renderDiffs(this.compareData.originalReport.xml, this.compareData.runResultReport.xml);
+      this.renderDiffs(
+        this.compareData.originalReport.xml,
+        this.compareData.runResultReport.xml,
+      );
       this.showReports();
     } else {
       this.router.navigate([DebugComponent.ROUTER_PATH]);
     }
   }
 
+  protected syncLeftAndRight(): void {
+    this.leftNode =
+      this.compareTreeComponent.leftTree.getSelected().originalValue;
+    this.rightNode =
+      this.compareTreeComponent.rightTree.getSelected().originalValue;
+    if (ReportUtil.isReport(this.leftNode)) {
+      this.leftReport = { ...this.leftNode };
+    }
+    if (ReportUtil.isReport(this.rightNode)) {
+      this.rightReport = { ...this.rightNode };
+    }
+    this.showDifference();
+  }
+
+  protected showDifference(): void {
+    if (this.leftNode && this.rightNode) {
+      const leftSide: string = this.extractMessage(this.leftNode);
+      const rightSide: string = this.extractMessage(this.rightNode);
+      this.renderDiffs(leftSide, rightSide);
+    }
+  }
+
+  protected changeNodeLinkStrategy(): void {
+    if (this.compareData && this.nodeLinkStrategy) {
+      localStorage.setItem(
+        `${this.compareData.viewName}.NodeLinkStrategy`,
+        this.nodeLinkStrategy,
+      );
+    }
+  }
+
+  protected changeView(view: View): void {
+    if (this.leftReport?.storageName && this.rightReport?.storageName) {
+      this.hideOrShowCheckpoints(
+        view,
+        this.leftReport.storageName,
+        this.leftReport.storageId,
+        this.compareTreeComponent.leftTree.elements.toArray(),
+      );
+      this.hideOrShowCheckpoints(
+        view,
+        this.rightReport.storageName,
+        this.rightReport.storageId,
+        this.compareTreeComponent.rightTree.elements.toArray(),
+      );
+    }
+    this.currentView = view;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```

Comment:
Dit kan toch `:void | undefined` zijn of zoiets dergelijks?

Reply:
Wat betekent dat? De functie kan iets retourneren dat gelijk is aan undefined, of er kan geen waarde geretourneerd worden, dus ook niet iets dat je kunt vergelijken met `undefined`. Vaag, maar ik snap dat dit mooier is dan een ESLint disable.

Ik stel voor om deze ignore wel te laten staan bij method FilterSideDrawerComponent.getTooltipSuggestion.getTooltipSuggestion() in file src/app/debug/filter-side-drawer/filter-side-drawer.component.ts. Deze method retourneert iets dat veel verschillende types kan hebben.